src/components/buttons/Button.jsx

```diff
@@ -0,0 +1,171 @@
+   * @see type="link-button-inverted" https://styleguide.brainly.com/latest/docs/interactive.html?type="link-button-inverted"#buttons
+   * @see type="link-button-peach" https://styleguide.brainly.com/latest/docs/interactive.html?type="link-button-peach"#buttons
+   * @see type="link-button-mustard" https://styleguide.brainly.com/latest/docs/interactive.html?type="link-button-mustard"#buttons
+   * @see type="destructive" https://styleguide.brainly.com/latest/docs/interactive.html?type="link-button-inverted"#buttons
+   * @see type="warning" https://styleguide.brainly.com/latest/docs/interactive.html?type="link-button-inverted"#buttons
+   * @see type="facebook" https://styleguide.brainly.com/latest/docs/interactive.html?type="link-button-inverted"#buttons
+   */
+  type: ButtonTypeType,
+  /**
+   * You can render icon inside each type of button on the left side
+   * @example <Button
+   *           icon={<Icon type="facebook" color="light" size={24} />}
+   *           buttonType="facebook"
+   *          >
+   *            Login with Facebook
+   *          </Button>
+   */
+  icon?: Node,
+  /**
+   * There are three sizes options for buttons, not need to be specify, default is medium
+   * @example <Button type="primary" size="medium">
+   *            button
+   *          </Button>
+   * @see size="small" https://styleguide.brainly.com/latest/docs/interactive.html?size="small"#buttons
+   * @see size="medium" https://styleguide.brainly.com/latest/docs/interactive.html?size="medium"#buttons
+   * @see size="large" https://styleguide.brainly.com/latest/docs/interactive.html?size="large"#buttons
+   */
+  size?: ButtonSizeType,
+  /**
+   * Specify href for button, optional string
+   * @example <Button href="https://brainly.com/" size="medium" type="primary-blue">
+   *            button
+   *          </Button>
+   */
+  href?: string,
+  /**
+   * Optional boolean for disabled button
+   * @example <Button type="primary-mint" disabled>
+   *            button
+   *          </Button>
+   */
+  disabled?: boolean,
+  /**
+   * Optional boolean for full width button
+   * @example <Button type="primary-mint" fullWidth>
+   *            button
+   *          </Button>
+   */
+  fullWidth?: boolean,
+  /**
+   * Additional class names
+   */
+  className?: string
+};
+
+const Button = ({
+  size,
+  type,
+  icon,
+  href,
+  fullWidth,
+  disabled,
+  children,
+  className,
+  ...props
+}: ButtonPropsType) => {
+  const btnClass = cx(
+    'sg-button', {
+      [`sg-button--${String(size)}`]: size,
+      [`sg-button--${String(type)}`]: type,
+      'sg-button--disabled': disabled,
+      'sg-button--full-width': fullWidth
+    },
+    className
+  );
+
+  const iconClass = cx(
+    'sg-button__icon', {
+      [`sg-button__icon--${size || ''}`]: size
+    },
+    className
+  );
+
+  const ico = icon !== undefined ? <span className={iconClass}>{icon}</span> : null;
```

Comment:
we should check for `null` as well i believe, otherwise `span` will be rendered as well (`Node` type is nullable by default, so it's still possible to pass `null` there)